Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpUnitTestsRunnerExecutionStrategy.cs

```diff
@@ -0,0 +1,226 @@
+                            }
+                        }
+
+                        var checkerResult = checker.Check(test.Input, message, test.Output, test.IsTrialTest);
+                        testResult.ResultType = checkerResult.IsCorrect ? TestRunResultType.CorrectAnswer : TestRunResultType.WrongAnswer;
+                        testResult.CheckerDetails = checkerResult.CheckerDetails;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(processExecutionResult), "Invalid ProcessExecutionResultType value.");
+                }
+
+                // Cleanup the .cs with the tested code to prepare for the next test
+                File.Delete(testedCodePath);
+                result.TestResults.Add(testResult);
+                count++;
+            }
+
+            return result;
+        }
+
+        private void ExtractTestResult(string receivedOutput, ref int passedTests, ref int totalTests)
+        {
+            Regex testResultsRegex =
+                new Regex(
+                    @"Test Count: (\d+), Passed: (\d+), Failed: (\d+), Warnings: \d+, Inconclusive: \d+, Skipped: \d+");
+            var res = testResultsRegex.Match(receivedOutput);
+            totalTests = int.Parse(res.Groups[1].Value);
+            passedTests = int.Parse(res.Groups[2].Value);
+        }
+
+        private void CorrectProjectReferences(Project project)
+        {
+            // Remove the first Project Reference (this should be the reference to the tested project)
+            var projectReference = project.GetItems("ProjectReference").FirstOrDefault();
+            if (projectReference != null)
+            {
+                project.RemoveItem(projectReference);
+            }
+
+            // Add a reference to tested code as a .cs file
+            project.AddItem("Compile", TestedCode);
+
+            // Remove previous NUnit reference (the path is probably pointing to the users package folder)
+            var nUnitPrevReference = project.Items.FirstOrDefault(x => x.EvaluatedInclude.Contains("nunit.framework"));
+            if (nUnitPrevReference != null)
+            {
+                project.RemoveItem(nUnitPrevReference);
+            }
+
+            // Add our NUnit Reference, if private is false, the .dll will not be copied and the tests will not run
+            Dictionary<string, string> nUnitMetaData = new Dictionary<string, string>();
+            nUnitMetaData.Add("Private", "True");
+            project.AddItem(
+                "Reference",
+                "nunit.framework, Version=3.6.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb, processorArchitecture=MSIL",
+                nUnitMetaData);
+    
+            // If we use NUnit we don't really need the VSTT, it will save us copying of the .dll
+            var vsTestFrameworkReference =
+                project.Items.FirstOrDefault(
```

Comment:
According the convention `project.Items` should be on the upper line, `.FirstOrDefault(x =>` on new one and the rest on new line too.